mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -4122,6 +4122,10 @@ VectorType mlir::vector::inferTransferOpMaskType(VectorType vecType,
   assert(invPermMap && "Inversed permutation map couldn't be computed");
   SmallVector<int64_t, 8> maskShape = invPermMap.compose(vecType.getShape());
 
+  // Turn a 0-D mask into a single-element 1-D mask.
```

Comment:
[nit] Personally, I'd appreciate a note saying that ATM `vector.mask` does not support 0-D masks (enforced [here](https://github.com/llvm/llvm-project/blob/4b50ec43d03d9ba9b43edd9a4743951f6498b964/mlir/include/mlir/Dialect/Vector/IR/VectorOps.td#L2509-L2510)). And that's basically the reason to "upgrade" this mask (which to me is a very good reason)